gobblin-aws/src/main/java/gobblin/aws/CloudInitScriptBuilder.java

```diff
@@ -0,0 +1,254 @@
+package gobblin.aws;
+
+import java.io.File;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Splitter;
+
+import gobblin.annotation.Alpha;
+import gobblin.cluster.GobblinClusterConfigurationKeys;
+import static gobblin.aws.GobblinAWSUtils.encodeBase64;
+
+
+/**
+ * Class to generate script for launching Gobblin cluster master and workers via cloud-init
+ * on EC2 instance boot up.
+ *
+ * @author Abhishek Tiwari
+ */
+@Alpha
+public class CloudInitScriptBuilder {
```

Comment:
Yes, but cloudinit / userdata has strict limits on number of characters allowed for the script - which gets really bloated after base64 conversion (the only format AWS api accepts it in) 
So, I first had the same model but was forced to move embedded mode to save char size space. The options were to do it like it is done right now or do string substitutions on a template. Since in gobblin-yarn we follow the first approach, I just stuck to it. 

However, this will eventually be changed into a bash script with variable substitution like you propose after we have pre-baked AMI with most of the required jars already baked into the instance (coming soon :) )